enumerables.rb

```diff
@@ -40,12 +75,24 @@ def my_map(&block)
       map_arr.push(block.call(i))
     end
     map_arr
+    to_enum
   end
 
-  def my_inject(thread, &block)
-    my_each do |item|
-      thread = block.call(thread, item)
+  def my_inject(accumulator = nil, operator = nill, &block)
+    if !block
+      if operator.nil?
+        operator = accumulator
+        acumulator = nil
+      end
+      operator.to_sym
+      each { |itm| accumulator = accumulator.nil? ? itm : accumulator.send(operator, itm) }
+    else
+      each { |itm| accumulator = accumulator.nil? ? itm : block.call(accumulator, itm) }
     end
-    thread
+    accumulator
+  end
+
+  def multiply_els(array = [])
+    array.my_inject(:*)
   end
```

Comment:
- [ ] Since this is not an actual enumerable, define the method outside module Enumerable
- [ ]  enumerables #multiply_els accepts an array as an argument and multiplies all the elements of the array together using #my_inject
      Failure/Error: actual = multiply_els [2, 4, 5]